src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -298,8 +298,8 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
         }
         RenderLayer& renderLayer = renderItem.layer;
         renderLayer.setRenderTiles(renderItem.source->getRenderTiles(), updateParameters.transformState);
-        if (const RenderLayerSymbolInterface* symbolLayer = renderLayer.getSymbolInterface()) {
```

Comment:
Do I understand correctly that layersNeedPlacement would have less items in container than renderItemsWithSymbols?

Reply:
less or equal :) several symbol layer can share the same bucket - if belong to the same layout group. We need to place buckets, so group leaders layers only are enough.